### DISCUSSION TITLE
v2 install issues, help

### DISCUSSION THREAD

**Discussion by Grimlock86** [gtxaspec/wz_mini_hacks#104] [General]
Running into an issue with the v2 install. I am running the compile_image.sh in terminal but it says no such directory exists and  it cannot  stat v2 kernal bin, no such directory exists, but it's there in the zip i downloaded. says demo.bin ready copy to sd card, but no demo.bin is found.
![Screenshot from 2022-06-18 13-02-18](https://user-images.githubusercontent.com/107766660/174449107-396ee05e-1c25-4433-8d34-095f66b845a1.png)

**Comment by gtxaspec** [comment]
what operating system are you using?

**Reply by Grimlock86** [reply]
Linux mint.

**Reply by gtxaspec** [reply]
edit the compile_image.sh

right under `#!/bin/bash`, add
`set -x`
save the file, 
then run the script, and share the output here

**Reply by Grimlock86** [reply]
<pre><font color="#729FCF"><b>~/Desktop/wz_mini_hacks-master/v2_install</b></font>$ ./compile_image.sh
+ UA_STRING=&apos;Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15&apos;
++ wget &apos;--header=Accept: text/html&apos; &apos;--user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15&apos; -qO- https://support.wyze.com/hc/en-us/articles/360024852172-Release-Notes-Firmware
++ grep -Eo &apos;(http|https)://[a-zA-Z0-9./?=_-]*&apos;
++ grep /v2/
++ sort -t . -k2r,5
++ head -1
++ grep demo
+ DL_URL=
+ echo &apos;create work directory&apos;
create work directory
+ mkdir v2_ro
+ echo &apos;download latest firmware&apos;
download latest firmware
+ wget -P ./v2_ro/
wget: missing URL
Usage: wget [OPTION]... [URL]...

Try `wget --help&apos; for more options.
+ echo &apos;extracting firmware to workdir&apos;
extracting firmware to workdir
+ unzip v2_ro/demo_4.9.8.1002.zip -d ./v2_ro/
unzip:  cannot find or open v2_ro/demo_4.9.8.1002.zip, v2_ro/demo_4.9.8.1002.zip.zip or v2_ro/demo_4.9.8.1002.zip.ZIP.
+ echo &apos;create firmware dir&apos;
create firmware dir
+ mkdir v2_ro/tmp_dir
+ echo &apos;unpack stock firmware&apos;
unpack stock firmware
+ ./fw_tool.sh unpack v2_ro/demo_4.9.8.1002.bin v2_ro/tmp_dir
dd: failed to open &apos;v2_ro/demo_4.9.8.1002.bin&apos;: No such file or directory
md5sum: v2_ro/tmp_dir/kernel.bin: No such file or directory
dd: failed to open &apos;v2_ro/demo_4.9.8.1002.bin&apos;: No such file or directory
md5sum: v2_ro/tmp_dir/rootfs.bin: No such file or directory
dd: failed to open &apos;v2_ro/demo_4.9.8.1002.bin&apos;: No such file or directory
md5sum: v2_ro/tmp_dir/driver.bin: No such file or directory
stat: cannot stat &apos;v2_ro/demo_4.9.8.1002.bin&apos;: No such file or directory
./fw_tool.sh: 25: arithmetic expression: expecting primary: &quot;&quot;
+ echo &apos;replace kernel&apos;
replace kernel
+ cp v2_kernel.bin v2_ro/tmp_dir/kernel.bin
+ echo &apos;pack firmware with new kernel&apos;
pack firmware with new kernel
+ ./fw_tool.sh pack v2_ro/tmp_dir/ demo.bin
1+0 records in
1+0 records out
1 byte copied, 0.000134534 s, 7.4 kB/s
cat: v2_ro/tmp_dir//rootfs.bin: No such file or directory
cat: v2_ro/tmp_dir//driver.bin: No such file or directory
cat: v2_ro/tmp_dir//appfs.bin: No such file or directory
./fw_tool.sh: 37: mkimage: not found
++ md5sum demo.bin
md5sum: demo.bin: No such file or directory
+ echo &apos;md5sum is:  &apos;
md5sum is:  
+ rm -rf v2_ro
+ echo &apos;demo.bin ready.  Please copy demo.bin to your memory card&apos;
demo.bin ready.  Please copy demo.bin to your memory card
</pre>

**Comment by clee604** [comment]
I had the same issue before and forgot to report it, the download script fails to download from the wyze server with a 403 error.

I resolved this by manually creating a v2_ro folder in the v2_install folder, then putting the newest v2 firmware file (zip or extracted bin both work) from wyze into the v2_ro folder. If you run the script then, it'll still fail to download the file, but the rest of the script runs fine and generates the correct demo.bin file

@gtxaspec might be worthwhile to put in a way to manually specify the v2 firmware file path and then it isn't dependent on the download script working

**Reply by gtxaspec** [reply]
I'll update the script to be more fault tolerant and descriptive... thanks

**Reply by Grimlock86** [reply]
I did that, and it deletes the v2_ro folder and can't locate the demo.bin it generated

+ UA_STRING='Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15'
++ wget '--header=Accept: text/html' '--user-agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 12_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15' -qO- https://support.wyze.com/hc/en-us/articles/360024852172-Release-Notes-Firmware
++ grep -Eo '(http|https)://[a-zA-Z0-9./?=_-]*'
++ grep /v2/
++ sort -t . -k2r,5
++ grep demo
++ head -1
+ DL_URL=
+ echo 'create work directory'
create work directory
+ mkdir v2_ro
mkdir: cannot create directory ‘v2_ro’: File exists
+ echo 'download latest firmware'
download latest firmware
+ wget -P ./v2_ro/
wget: missing URL
Usage: wget [OPTION]... [URL]...

Try `wget --help' for more options.
+ echo 'extracting firmware to workdir'
extracting firmware to workdir
+ unzip v2_ro/demo_4.9.8.1002.zip -d ./v2_ro/
Archive:  v2_ro/demo_4.9.8.1002.zip
  inflating: ./v2_ro/demo_4.9.8.1002.bin  
  inflating: ./v2_ro/__MACOSX/._demo_4.9.8.1002.bin  
+ echo 'create firmware dir'
create firmware dir
+ mkdir v2_ro/tmp_dir
+ echo 'unpack stock firmware'
unpack stock firmware
+ ./fw_tool.sh unpack v2_ro/demo_4.9.8.1002.bin v2_ro/tmp_dir
2097152+0 records in
2097152+0 records out
2097152 bytes (2.1 MB, 2.0 MiB) copied, 4.96342 s, 423 kB/s
bee7951307dc53edf59599cd15294a88  v2_ro/tmp_dir/kernel.bin
3473408+0 records in
3473408+0 records out
3473408 bytes (3.5 MB, 3.3 MiB) copied, 8.44841 s, 411 kB/s
c4ac3804e50cae555a21d2afbc63f26a  v2_ro/tmp_dir/rootfs.bin
655360+0 records in
655360+0 records out
655360 bytes (655 kB, 640 KiB) copied, 1.54188 s, 425 kB/s
e9e8f7b18e57845bf6835a9c4b5788f1  v2_ro/tmp_dir/driver.bin
4849664+0 records in
4849664+0 records out
4849664 bytes (4.8 MB, 4.6 MiB) copied, 11.6576 s, 416 kB/s
50deec4b53dde03c3efe98535e2f5b16  v2_ro/tmp_dir/appfs.bin
+ echo 'replace kernel'
replace kernel
+ cp v2_kernel.bin v2_ro/tmp_dir/kernel.bin
+ echo 'pack firmware with new kernel'
pack firmware with new kernel
+ ./fw_tool.sh pack v2_ro/tmp_dir/ demo.bin
1+0 records in
1+0 records out
1 byte copied, 0.000126103 s, 7.9 kB/s
./fw_tool.sh: 37: mkimage: not found
++ md5sum demo.bin
md5sum: demo.bin: No such file or directory
+ echo 'md5sum is:  '
md5sum is:  
+ rm -rf v2_ro
+ echo 'demo.bin ready.  Please copy demo.bin to your memory card'
demo.bin ready.  Please copy demo.bin to your memory card

**Reply by clee604** [reply]
for me it generates the demo.bin in the v2_install folder
edit: it looks like mkimage didn't run. You may be missing some dependencies

If i remember correctly the package you need is uboot-mkimage

**Reply by gtxaspec** [reply]
the latest release includes an updated v2 installation script... you can place the demo_4.9.8.1002.zip archive in the same folder as the script and it will use the local archive instead of downloading.

**Comment by Vendo232** [comment]
after flashing the V2 I`m getting error " Can not find specified network name" any idea how to fix it.

When I flash official FW 1002 it works , when I flash patche 1002 it blinks orange/blue and when I try to add new camera it tells me  " Can not find specified network name" 

Check for T20 platform
+ '[' -b /dev/mtdblock10 ]
+ echo 'Found T20'
Found T20
+ echo 43
+ echo 'in'
+ sleep 3
+ '[' -d /sys/class/gpio/gpio38 ]
+ echo 38
+ echo out
+ echo 1
+ '[' -d /sys/class/gpio/gpio39 ]
+ echo 39
+ echo out
+ echo 1
+ mkdir -p /sdcard
+ mount -t vfat /dev/mmcblk0p1 /sdcard -o 'rw,umask=0000,dmask=0000'
+ '[' -f /sdcard/wz_mini/wz_mini.conf ]
+ '[' -f /sdcard/wz_mini/wz_mini.conf ]
+ initram_init
+ mkdir /wz
+ mount -t squashfs /dev/mtdblock2 /wz
+ '[['  '==' true ]]
+ '[' '!' -f /sdcard/wz_mini/etc/init.d/wz_init.sh ]
+ echo 'wz_mini not found, booting stock'
wz_mini not found, booting stock
+ mount --move /dev /wz/dev
+ mount --move /sys /wz/sys
+ mount --move /proc /wz/proc
+ cp /tmp/wz_mini_initramfs.log /sdcard/.wz_mini_initramfs.log

I guess it could be SD CARD at fault

**Reply by gtxaspec** [reply]
can you share wz_user.log?

**Reply by Vendo232** [reply]
OK issue closed, it was the SD CARD formatting software on PC/Windows

minitool partition wizard ( does not work )
guiformat.exe ( work )

both formatting sdcard to FAT32.

**Reply by dasb00ter** [reply]
@Vendo232 can I ask what size of card that was? I am struggling with the same thing on 64gb card...and thought it was maybe the size since there is apparently a limitation of 32gb for flashing and so I used an 8 gb and then tried to copy over the files to the 64gb and have the same situation with that cannot find specified network problem
